From 1a5dfed8f73e08e65afb0711bac29f9a63548595 Mon Sep 17 00:00:00 2001
From: Sergio Costas <raster@rastersoft.com>
Date: Fri, 30 Nov 2018 20:28:55 +0100
Subject: [PATCH 23/55] fileItem.js: fix undefined property error

When a new file is added to the desktop, it doesn't have the
"nautilus-icon-position" metadata entry. Unfortunately, in this case
the "this._savedCoordinates" is not defined, so when the code tries to
paint it, it fails.

This patch fixes this by defining it with NULL value when there are no
stored coordinates in the metadata.
---
 fileItem.js | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fileItem.js b/fileItem.js
index ff060fa..4f7e435 100644
--- a/fileItem.js
+++ b/fileItem.js
@@ -71,6 +71,8 @@ var FileItem = class {
         let savedCoordinates = fileInfo.get_attribute_as_string('metadata::nautilus-icon-position');
         if (savedCoordinates != null)
             this._savedCoordinates = savedCoordinates.split(',').map(x => Number(x));
+        else
+            this._savedCoordinates = null;
 
         this._state = State.NORMAL;
 
-- 
2.20.0

